### PR TITLE
[Merged by Bors] - feat(Positivity): Strictness extractors

### PR DESCRIPTION
**Pull request by YaelDillies**
Define combinators to turn strictness assumptions into proofs of nonnegativity or nonzeroness.

These are useful to write `positivity` extensions where having `0 < a` doesn't help.

We might want binary combinators in the future. Eg the pattern "Binary function `foo` is positive/nonnegative/nonzero if its inputs are" is very common.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by ericrbg** [comment]
I wonder if it's worth rewriting `solve` with these.

**Comment by YaelDillies** [comment]
I doubt it will shorten the code much since `solve` also includes logic about direction the inequality is in.

File: Mathlib/Tactic/Positivity/Core.lean [hunk 1]
```diff
@@ -49,6 +49,20 @@ def Strictness.toString : Strictness zα pα e → String
   | nonzero _ => "nonzero"
   | none => "none"
 
+/-- Combinator to turn a `Strictness` information about `e` into a proof that `e` is nonnegative, if
+possible. -/
```
Comment (eric-wieser) [inline Mathlib/Tactic/Positivity/Core.lean:53]: What makes this a combinator? Perhaps just
```suggestion
/-- Extract a proof that `e` is nonnegative, if possible, from `Strictness` information about `e`. -/
```

File: Mathlib/Tactic/Positivity/Core.lean [hunk 2]
```diff
@@ -49,6 +49,20 @@ def Strictness.toString : Strictness zα pα e → String
...
+def Strictness.toNonneg {e} : Strictness zα pα e → Option Q(0 ≤ $e)
+  | .positive pf => some q(le_of_lt $pf)
+  | .nonnegative pf => some pf
+  | _ => .none
+
+/-- Combinator to turn a `Strictness` information about `e` into a proof that `e` is nonzero, if
+possible. -/
```
Comment (YaelDillies) [inline Mathlib/Tactic/Positivity/Core.lean:60]:
```suggestion
/-- Extract a proof that `e` is nonzero, if possible, from `Strictness` information about `e`. -/
```

**Comment by eric-wieser** [comment]
bors d+

Thanks!

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**